tests/test_cc_uart.py

```diff
@@ -1,123 +1,107 @@
 ##########################################################################
-# Copyright (c) 2010-2022 Robert Bosch GmbH
+# Copyright (c) 2010-2023 Robert Bosch GmbH
 # This program and the accompanying materials are made available under the
 # terms of the Eclipse Public License 2.0 which is available at
 # http://www.eclipse.org/legal/epl-2.0.
 #
 # SPDX-License-Identifier: EPL-2.0
 ##########################################################################
 
-import pathlib
-import subprocess
-import sys
-import threading
-import time
 
 import pytest
 
 from pykiso import message
-from pykiso.lib.connectors.cc_uart import CCUart
+from pykiso.lib.connectors.cc_uart import CCUart, IncompleteCCMsgError
 
 
 @pytest.fixture
-def virtual_serial(tmp_path):
-    UART0 = tmp_path / "vt0"
-    UART1 = tmp_path / "vt1"
-    BAUD = 9600
-
-    uart0 = str(pathlib.Path(UART0).expanduser())
-    uart1 = str(pathlib.Path(UART1).expanduser())
-
-    print(f"running 'socat -d -d pty,link={uart0},rawer pty,link={uart1},rawer'")
-    socat = subprocess.Popen(
-        [
-            "/usr/bin/env",
-            "socat",
-            "-d",
-            "-d",
-            f"pty,link={uart0},rawer",
-            f"pty,link={uart1},rawer",
-        ],
-    )
-    time.sleep(1)  # wait for socat to be up and running
-    assert socat.returncode is None
-    yield uart0, uart1, BAUD
-    socat.terminate()
-
-
-def _communication_send_check(serial_port, baud):
-    """Test to do with HW connected or UART emulator"""
-    # message_to_send = message.Message(msg_type=message.MessageType.COMMAND, sub_type=message.MessageCommandType.TEST_CASE_SETUP, test_section=1, test_suite=1, test_case=1)
-    message_to_send = message.Message(
-        msg_type=message.MessageType.ACK,
-        sub_type=message.MessageAckType.ACK,
-        test_suite=2,
-        test_case=3,
-    )
-    # manually set msg token so it can be checked for
-    message_to_send.msg_token = 0
-    # Initialize and open uart
-    print(f"sending test message to {serial_port}")
-    ch = CCUart(serialPort=serial_port, baudrate=baud)
-    ch.open()
-    # Send message
-    ch._cc_send(message_to_send)
-    # Close uart
-    ch.close()
+def CCuart_instance():
+    return CCUart(serialPort="port")
+
+
+@pytest.fixture
+def serial_read_mock(mocker):
+    return mocker.patch("serial.Serial.read")
+
+
+def test_incomplete_ccmsg_error(caplog):
```

Comment:
you don't need the caplog for this method

Reply:
Done